runtime/drivers/duckdb/olap.go

```diff
@@ -181,6 +186,344 @@ func (c *connection) EstimateSize() (int64, bool) {
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:    fmt.Sprintf("ALTER TABLE %s.default ADD COLUMN %s %s", safeSQLName(dbName), safeSQLName(columnName), typ),
+		Priority: 1,
+	})
+	if err != nil {
+		return err
+	}
+
+	// recreate view to propagate schema changes
+	return c.Exec(ctx, &drivers.Statement{
+		Query:    fmt.Sprintf("CREATE OR REPLACE VIEW %s AS SELECT * FROM %s.default", safeSQLName(tableName), safeSQLName(dbName)),
+		Priority: 1,
+	})
+}
+
+// AlterTableColumn implements drivers.OLAPStore.
+func (c *connection) AlterTableColumn(ctx context.Context, tableName, columnName, newType string) error {
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("ALTER TABLE %s ALTER %s TYPE %s", safeSQLName(tableName), safeSQLName(columnName), newType),
+			Priority: 1,
+		})
+	}
+
+	version, exist, err := c.tableVersion(tableName)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		return fmt.Errorf("table %q does not exist", tableName)
+	}
+	dbName := dbName(tableName, version)
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:    fmt.Sprintf("ALTER TABLE %s.default ALTER %s TYPE %s", safeSQLName(dbName), safeSQLName(columnName), newType),
+		Priority: 1,
+	})
+	if err != nil {
+		return err
+	}
+
+	// recreate view to propagate schema changes
+	return c.Exec(ctx, &drivers.Statement{
+		Query:    fmt.Sprintf("CREATE OR REPLACE VIEW %s AS SELECT * FROM %s.default", safeSQLName(tableName), safeSQLName(dbName)),
+		Priority: 1,
+	})
+}
+
+// CreateTableAsSelect implements drivers.OLAPStore.
+func (c *connection) CreateTableAsSelect(ctx context.Context, name string, view bool, sql string) error {
+	if view {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("CREATE OR REPLACE VIEW %s AS (%s)", safeSQLName(name), sql),
+			Priority:    1,
+			LongRunning: true,
+		})
+	}
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("CREATE OR REPLACE TABLE %s AS (%s)", safeSQLName(name), sql),
+			Priority:    1,
+			LongRunning: true,
+		})
+	}
+	// create a new db file in /<instanceid>/<name> directory
+	sourceDir := filepath.Join(filepath.Dir(c.config.DBFilePath), name)
+	if err := os.Mkdir(sourceDir, fs.ModePerm); err != nil && !strings.Contains(err.Error(), "file exists") {
+		return err
+	}
+
+	// check if some older version existed previously to detach it later
+	oldVersion, oldVersionExists, _ := c.tableVersion(name)
+
+	newVersion := fmt.Sprint(time.Now().UnixMilli())
+	dbFile := filepath.Join(sourceDir, fmt.Sprintf("%s.db", newVersion))
+	db := dbName(name, newVersion)
+	// attach new db
+	err := c.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("ATTACH '%s' AS %s", dbFile, safeSQLName(db))})
```

Comment:
What if the DB file name contains single quotes?

Reply:
Will handle that. 
In general I think instead of handling all such special cases we should evaluate if we can make our lives easier and only allow alphanumeric characters and spaces in source, model, dashboard names. What difference is it going to make if a user must make their source `customer data` instead of `customer's data`

Reply:
Have seen some issues about this stuff in the past (mostly for column names though). It can sneak in – e.g. someone's tool exports a file named `Benjamin's export.csv` and then they drag+drop it into Rill. Doing sanitization of input names everywhere easily becomes as complex as handling complex names everywhere.

Even if we add upstream restrictions, I find it better to just handle it at every layer instead of needing to think through how those assumptions propagate. Kind of like layers of security.